feat(root): use utxolib toOutputScriptTryFormats in isValidAddress

rushilbg opened the pull request:
BG-60666

BREAKING CHANGE: this drops support for the legacy Litecoin format.
Litecoin addresses starting with 3 will not be supported anymore.

<!--
# Please be aware of the following when making your pull request:

## Description

Please include a summary of your proposed changes and which issue is being addressed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Issue Number

Internal Users - Please include the related internal tracking number (e.g. BG-000000).
External Users - Please link to any relevant github issues as necessary.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code compiles correctly for both Node and Browser environments
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [ ] The ticket or github issue was included in the commit message as a reference
- [ ] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
-->

OttoAllmendinger committed:
feat(utxo-lib): simplify isValidAddress

OttoAllmendinger commented on modules/abstract-utxo/src/config.ts:
```diff
@@ -1,5 +1,5 @@
-// Supported cross-chain recovery routes. The coin to be recovered is the index, the valid coins for recipient wallets
-// are listed in the array.
+// Supported cross-chain recovery routes. The object index is the coin to be recovered, and the value is
+// an array of valid coins for recipient wallets of the original accidental/unintentional deposit.
```
let's do this in a separate PR

rushilbg replied:
Removed

OttoAllmendinger reviewed (changes requested):
you should call the PR title "use utxolib toOutputScriptTryFormats in `isValidAddress`"

The fact that ecash addresses are now supported is more of a side effect (I believe this PR also enables cashaddr support for coin `bch`)

rushilbg committed:
feat(sdk-coin-ltc): remove canonicalising LTC addresses

BG-60666

BREAKING CHANGE: this drops support for the legacy Litecoin format.
Litecoin addresses starting with 3 will not be supported anymore.

rushilbg committed:
refactor(root): remove unused getAddressPrefix fn from BCH/BSV

BG-60666

OttoAllmendinger commented:
Background: we moved all address parsing logic, including support for nonstandard formats, to `utxolib.addressFormat` a while ago. BitGoJS still implemented custom address validation in `isValidAddress()` which is slightly inconsistent.

This change now has these changes for `isValidAddress()` 

* rejects legacy litecoin addresses (3-prefixed)
* accept `bitcoincash:` cashaddrs for the `bch` coin
* accept `ecash:` cashaddrs for the `bcha` coin

OttoAllmendinger commented on modules/abstract-utxo/src/abstractUtxoCoin.ts:
```diff
@@ -292,72 +292,21 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
    * @param address
    * @param forceAltScriptSupport
    */
   isValidAddress(address: string, forceAltScriptSupport = false): boolean {
```
let's put support for non-defined formats behind a parameter

```suggestion
  isValidAddress(address: string, param?: { anyFormat: boolean } | /* legacy parameter */ boolean): boolean {
    if (typeof param === 'boolean' && param) {
      throw new Error('deprecated');
    }
    
    const formats = (param && param.anyFormat) ? ['default'] : undefined;
    
     try {
        utxolib.addressFormat.toOutputScriptTryFormats(address, this.network, formats);
        return true;
      } catch (e) {
        return false;
      }
  }  
```

rushilbg committed:
feat(abstract-utxo): allow for override in isValidAddress

BG-60666: will also use this override in backupKeyRecovery
